api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/AllergyIntoleranceTranslatorImpl.java

```diff
@@ -108,14 +114,13 @@ public Allergy toOpenmrsType(Allergy allergy, AllergyIntolerance fhirAllergy) {
 		allergy.setUuid(fhirAllergy.getId());
 		if (allergy.getAllergen() == null) {
 			Allergen allergen = new Allergen();
-			if (!fhirAllergy.getCode().getCoding().isEmpty()) {
-				allergen.setCodedAllergen(
-				    conceptService.getConceptByUuid(fhirAllergy.getCode().getCoding().get(0).getCode()).orElse(null));
+			if (fhirAllergy.hasCode()) {
+				allergen.setCodedAllergen(conceptTranslator.toOpenmrsType(fhirAllergy.getCode()));
 				allergen.setNonCodedAllergen(fhirAllergy.getCode().getCoding().get(0).getDisplay());
 			}
```

Comment:
If the code doesn't exist as a concept in Openmrs, It will return null and set NonCodedAllergen to coding display, which is okay but what if code exists? Meaning Both codedAllergen and NonCodedAllergen will have a value. I think NonCodedAllergen should only have a value when a concept doesn't exist that allergen. I might be wrong

Reply:
I'm also unsure about their later but I'll ask this on talk for clarifications